### PR TITLE
add alexnet/squeezenet/wideresnet backbone & pytorch converter

### PR DESCRIPTION
**Pull request by diamond0910**

### CODE REVIEW

File: python/jittor/test/test_models.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    
+
+skip_this_test = False
+
+
+@unittest.skipIf(skip_this_test, "skip_this_test")
+class TestModels(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.testmodels = [
```
Comment (Gword) [inline python/jittor/test/test_models.py]: Note the naming convention, Modify testmodels to test_models

File: python/jittor/test/test_models.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            # Set eval to avoid dropout layer
+            pytorchmodel.eval()
+            jittormodel.eval()
+            # Jittor loads pytorch parameters to ensure forward alignment
+            jittormodel.load_parameters(pytorchmodel.state_dict())
+            # Judge pytorch & jittor forward relative error. If the differece is lower than threshold, this test passes.
+            pytorchresult = pytorchmodel(pytorchtestimg)
+            jittorresult = jittormodel(jittortestimg)
+            x = pytorchresult.detach().cpu().numpy() + 1
+            y = jittorresult.data + 1
```
Comment (Gword) [inline python/jittor/test/test_models.py]: Note the naming convention, use "_" to separate words

File: python/jittor/test/test_models.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+#     Wenyang Zhou <576825820@qq.com>
+#     Dun Liang <randonlang@gmail.com>. 
+# All Rights Reserved.
+# This file is subject to the terms and conditions defined in
+# file 'LICENSE.txt', which is part of this source code package.
+# ***************************************************************
+import numpy as np
+import unittest
+import jittor as jt
+import numpy as np
```
Comment (Gword) [inline python/jittor/test/test_models.py]: delete this

File: python/jittor/test/test_pytorch_converter.py [hunk 4]
```diff
@@ -5,325 +5,260 @@
...
+        out = self.conv1(x)
+        out = self.bn1(out)
+        out = self.relu(out)
+ 
+        out = self.conv2(out)
+        out = self.bn2(out)
+ 
+        if self.downsample is not None:
+            residual = self.downsample(x)
+
```
Comment (Gword) [inline python/jittor/test/test_pytorch_converter.py]: delete useless blank line

File: python/jittor/test/test_pytorch_converter.py [hunk 5]
```diff
@@ -5,325 +5,260 @@
...
+        out = self.conv2(out)
+        out = self.bn2(out)
+        out = self.relu(out)
+ 
+        out = self.conv3(out)
+        out = self.bn3(out)
+ 
+        if self.downsample is not None:
+            residual = self.downsample(x)
+
```
Comment (Gword) [inline python/jittor/test/test_pytorch_converter.py]: delete useless blank line

File: python/jittor/test/test_pytorch_converter.py [hunk 6]
```diff
@@ -5,325 +5,260 @@
...
+import random
+
+# setup random seed
+def setup_seed(seed):
+    np.random.seed(seed)
+    random.seed(seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = True
```
Comment (Gword) [inline python/jittor/test/test_pytorch_converter.py]: unused function?

File: python/jittor/models/squeezenet.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+            nn.Dropout(p=0.5), 
+            final_conv, 
+            nn.Relu(), 
+            nn.Pool(13, op="mean")
+        )
+        for m in self.modules():
+            if isinstance(m, nn.Conv):
+                if (m is final_conv):
+                    init.gauss_(m.weight, mean=0.0, std=0.01)
+                # else:
```
Comment (cxjyxxme) [inline python/jittor/models/squeezenet.py]: remove

File: python/jittor/test/test_models.py [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    def test_models(self):
+        threshold = 1e-3
+        # Define numpy input image
+        bs = 1
+        turns = 100
+        testimg = np.random.random((bs,3,224,224)).astype('float32')
+        # Define pytorch & jittor input image
+        pytorchtestimg = torch.Tensor(testimg).cuda()
+        jittortestimg = jt.array(testimg)
+        for testmodel in self.testmodels:
```
Comment (cxjyxxme) [inline python/jittor/test/test_models.py]: Note the naming convention, testimg to test_img, pytorchtestimg to pytorch_test_img, jittor_Test_img, test_models etc.

File: python/jittor/test/test_models.py (hunk 2 above)
Comment (cxjyxxme) [inline python/jittor/test/test_models.py]: Note the naming convention

File: python/jittor/utils/pytorch_converter.py [hunk 9]
```diff
@@ -1,288 +1,146 @@
...
 # file 'LICENSE.txt', which is part of this source code package.
 # ***************************************************************
-import sys
-import contextlib
-import os
-import signal
-import jittor as jt
-jt.dirty_fix_pytorch_runtime_error()
-import torch
+import ast, astunparse
```
Comment (cxjyxxme) [inline python/jittor/utils/pytorch_converter.py]: add to  Dependency package？

File: python/jittor/test/test_pytorch_converter.py [hunk 10]
```diff
@@ -5,325 +5,260 @@
...
-), 'x': input_0} end
-root in:[] out:[] end"""
-        assert str(ct) == ans
-        code = ct.to_jt()
-        lc = {}
-        exec(code, globals(), lc)
-        print(code)
+@unittest.skipIf(torch is None, "pytorch not found.")
+class TestPytorchConverter(unittest.TestCase):
+    def test_pytorch_converter(self):
```
Comment (cxjyxxme) [inline python/jittor/test/test_pytorch_converter.py]: add more model test cases? resnet108 vgg alexnet wideresnet squeezenet etc.

**Review by cxjyxxme — CHANGES_REQUESTED** [review]
GJ!

File: python/jittor/test/test_pytorch_converter.py (hunk 10 above)
Comment (Gword) [reply python/jittor/test/test_pytorch_converter.py]: No need to test so many models? if only to test converter

Comment (Gword) [reply python/jittor/test/test_pytorch_converter.py]: > add more model test cases? resnet108 vgg alexnet wideresnet squeezenet etc.

No need to test so many models? if only to test converter

**Pull request closed without merging**